src/compile/data/nullfilter.ts

```diff
@@ -31,38 +31,28 @@ export namespace nullFilter {
     }, {});
   }
 
-  export const parseUnit = parse;
-
-  export function parseFacet(model: FacetModel) {
-    let nullFilterComponent = parse(model);
+  export function mergeIfCompatible(dataComponent: DataComponent, childDataComponents: DataComponent[]) {
+    const nullFilterComponent = childDataComponents.reduce((collector, childData) => {
```

Comment:
I don't understand why you have to merge this one first here.  

I think you only need to check if all of children are compatible.  If so, we can merge them into the parent.